app/internal/import_file.py

```diff
@@ -0,0 +1,185 @@
+        except ValueError:
+            return False
+    else:
+        check_date = date
+    if check_date.year > now_year + valid_dates or \
+       check_date.year < now_year - valid_dates:
+        return False
+    return True
+
+
+def is_event_text_valid(row: str) -> bool:
+    """Check if the row contains valid data"""
+    get_values = re.findall(r"^(\w{" + str(EVENT_HEADER_NOT_EMPTY) + "," +
+                            str(EVENT_HEADER_LIMIT) + r"}),\s(\w{0," +
+                            str(EVENT_CONTENT_LIMIT) +
+                            r"}),\s(\d{2}-\d{2}-\d{4})," +
+                            r"\s(\d{2}-\d{2}-\d{4})$", row)
+    if get_values:
+        if len(get_values[0]) == NUM_OF_VALUES:
+            return True
+    return False
+
+
+def is_file_valid_to_import(file: str) -> bool:
+    """
+    checking before importing that the file exist, the file extension and
+    the size meet the rules we have set.
+    """
+    if not is_file_exist(file):
+        return False
+    if not is_file_extension_valid(file):
+        return False
+    if not is_file_size_valid(file):
+        return False
+    return True
+
+
+def is_file_valid_to_save_to_database(events: List[Dict[str, Union[str, Any]]],
+                                      max_event_start_date: int
+                                      = MAX_EVENTS_START_DATE) -> bool:
+    """
+    checking after importing that there is no larger quantity of events
+    with the same date according to the rule we have set.
+    """
+    same_date_counter = 1
+    date_n_count = {}
+    for event in events:
+        if event["S_Date"] in date_n_count:
+            date_n_count[event["S_Date"]] += 1
+            if date_n_count[event["S_Date"]] > same_date_counter:
+                same_date_counter = date_n_count[event["S_Date"]]
+        else:
+            date_n_count[event["S_Date"]] = 1
+    return same_date_counter <= max_event_start_date
+
+
+def import_txt_file(txt_file: str) -> List[Dict[str, Union[str, Any]]]:
+    calendar_content = []
+    with open(txt_file, "r") as text:
+        events = text.readlines()
+    for event in events:
+        if not is_event_text_valid(event):
+            return list()
+        head, content, start_date, end_date = event.split(", ")
+        if not is_date_in_range(start_date) or \
+           not is_date_in_range(end_date.replace("\n", "")):
+            return list()
+        start_date = datetime.datetime.strptime(start_date, "%m-%d-%Y")
+        end_date = datetime.datetime.strptime(end_date.replace("\n", ""),
+                                              "%m-%d-%Y")
+        calendar_content.append({"Head": head,
+                                 "Content": content,
+                                 "S_Date": start_date,
+                                 "E_Date": end_date})
+    return calendar_content
+
+
+def import_ics_file(ics_file: str) -> List[Dict[str, Union[str, Any]]]:
```

Comment:
A bit of a long, indented function. It is have more than one role (opening file, creating objects, creating dictionary, walking inside etc). Consider splitting it.

Reply:
this function split to "open_ics", "is_valid_data_event_ics" and "save_calendar_content_ics" to support the main function